backend/src/user/router.py

```diff
@@ -51,11 +58,16 @@ def create_user(user: UserAuthSchema, db: Session = Depends(session_opener)):
     :rtype: User
     """
     hashed_password = pwd_context.hash(user.password)
-    db_user = User(username=user.username, hashed_password=hashed_password)
-    db.add(db_user)
-    db.commit()
-    db.refresh(db_user)
-    return db_user
+    try:
+        db_user = User(username=user.username, hashed_password=hashed_password)
+        db.add(db_user)
+        db.commit()
+        db.refresh(db_user)
+        return db_user
```

Comment:
`db.commit()` is the only function that will throw exception, you can just catch this function, rest of them are not required to do so.